src/Dictionaries/CacheDictionary.cpp

```diff
@@ -575,37 +586,38 @@ void CacheDictionary::setAttributeValue(Attribute & attribute, const Key idx, co
     switch (attribute.type)
     {
         case AttributeUnderlyingType::utUInt8:
-            std::get<ContainerPtrType<UInt8>>(attribute.arrays)[idx] = value.get<UInt64>();
+            /// Strange code. Why UInt8 and UInt64? I looked through the history. It's always been like this.
```

Comment:
`Field` uses UInt64 as the underlying storage type for smaller ints, but I think `get<UInt8>` would also work. By the way, why are you switching to `safeGet` here? It is a method of limited utility that throws a generic (bad) error message, mostly used for user input. `get` has an assertion inside, that should for checks of internal logic.

Reply:
I was surprised about UInt64 about underlying type in Field. I know that get<UInt8> works, but get<Float32> doesn't work. Do you know why? P.S. I've added safeGet only for debug.

Reply:
> get works, but get doesn't work.

Looks like a typo :) What do you mean?

Reply:
`get` and `safeGet` might have different handling of signedness -- `safeGet` seems to check signedness, and `get` ignores it. I tried to make signedness strict once, but there are hundreds of places that don't respect that, and the benefit is marginal, so I gave up.

Reply:
> > get works, but get doesn't work.
> 
> 
> 
> Looks like a typo :) What do you mean?

LOL. I meant getUInt8 and getFloat32. This may be due to the internal representation of Float. In general, this is strange, it turns out that we do not have strict type checking.